server/sonar-server/src/main/java/org/sonar/server/rule/RuleUpdater.java

```diff
@@ -264,34 +273,85 @@ private void deleteOrUpdateParameters(DbSession dbSession, RuleUpdate update, Ru
 
       if (value != null) {
         // Update linked active rule params or create new one
-        for (ActiveRuleDto activeRuleDto : activeRules.get(customRule)) {
-          for (ActiveRuleParamDto activeRuleParamDto : activeRuleParams.get(activeRuleDto)) {
-            if (activeRuleParamDto.getKey().equals(key)) {
-              dbClient.activeRuleDao().updateParam(dbSession, activeRuleDto, activeRuleParamDto.setValue(value));
-            } else {
-              dbClient.activeRuleDao().insertParam(dbSession, activeRuleDto, ActiveRuleParamDto.createFor(ruleParamDto).setValue(value));
-            }
-          }
-        }
+        from(activeRuleParamsByActiveRule.keySet())
```

Comment:
I think it's a very weird pattern to use list transformations to update datastores